pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -393,33 +348,9 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 							severityAlertLabelKey: "warning",
 						},
 					},
-					{
-						Record: "vec_by_virt_handlers_all_client_rest_requests_in_last_5m",
-						Expr: intstr.FromString(
-							generateValueDifferenceInFiveMinutesQuery(generateAllPodRequestsQuery("virt-handler", ns, "")),
-						),
-					},
-					{
-						Record: "vec_by_virt_handlers_all_client_rest_requests_in_last_hour",
-						Expr: intstr.FromString(
-							generateValueDifferenceInHourQuery(generateAllPodRequestsQuery("virt-handler", ns, "")),
-						),
-					},
-					{
-						Record: "vec_by_virt_handlers_failed_client_rest_requests_in_last_5m",
-						Expr: intstr.FromString(
-							generateValueDifferenceInFiveMinutesQuery(generateAllPodRequestsQuery("virt-handler", ns, "(4|5)[0-9][0-9]")),
-						),
-					},
-					{
-						Record: "vec_by_virt_handlers_failed_client_rest_requests_in_last_hour",
-						Expr: intstr.FromString(
-							generateValueDifferenceInHourQuery(generateAllPodRequestsQuery("virt-handler", ns, "(4|5)[0-9][0-9]")),
-						),
-					},
 					{
 						Alert: "VirtHandlerRESTErrorsHigh",
-						Expr:  intstr.FromString("(vec_by_virt_handlers_failed_client_rest_requests_in_last_hour / vec_by_virt_handlers_all_client_rest_requests_in_last_hour) >= 0.05"),
```

Comment:
Just a thought: may be it is better to keep all these records (vec_by_virt_handlers_failed_client_rest_requests_in_last_hour, vec_by_virt_handlers_all_client_rest_requests_in_last_hour, etc...)

In your case you stores only error ratio, but for debugging it might be helpful to check separate metrics: failed and total

Reply:
Hey Denys,
Thanks for reviewing.
I agree that these recording records can be useful for debugging but Recording rules are for pre-calculating frequently used or computationally expensive queries. 
They require CPU resources and in our case we never use them in any of our alerts,
Because of counters resets they might show wrong value and mislead .